hystrix-contrib/hystrix-javanica/hystrix-javanica-core/src/main/java/com/netflix/hystrix/contrib/javanica/aop/AbstractCommandMetaHolderBuilder.java

```diff
@@ -0,0 +1,220 @@
+		if (hystrixCommand == null) {
+			throw new IllegalStateException("batch method must be annotated with HystrixCommand annotation");
+		}
+		// method of batch hystrix command must be passed to metaholder because basically collapser doesn't have any actions
+		// that should be invoked upon intercepted method, it's required only for underlying batch command
+
+		builder.args(args);
+		builder.method(batchCommandMethod);
+		builder.obj(obj);
+		builder.objectClass(proxyClass);
+		builder.hystrixCollapser(hystrixCollapser);
+		builder.defaultCollapserKey(method.getName());
+		builder.collapserExecutionType(ExecutionType.getExecutionType(collapserReturnType));
+
+		builder.defaultCommandKey(batchCommandMethod.getName());
+		builder.hystrixCommand(hystrixCommand);
+		builder.executionType(ExecutionType.getExecutionType(batchReturnType));
+		builder.observable(observable);
+		FallbackMethod fallbackMethod = MethodProvider.getInstance().getFallbackMethod(proxyClass, batchCommandMethod);
+		if (fallbackMethod.isPresent()) {
+			fallbackMethod.validateReturnType(batchCommandMethod);
+			builder.fallbackMethod(fallbackMethod.getMethod()).fallbackExecutionType(ExecutionType.getExecutionType(fallbackMethod.getMethod().getReturnType()));
+		}
+		customizeCollapserBuilder(hystrixCollapser, batchCommandMethod);
+
+	}
+
+	protected void customizeCollapserBuilder(HystrixCollapser hystrixCollapser, Method batchCommandMethod) {
+
+	}
+
+	protected void customizeCommandBuilder(HystrixCommand hystrixCommand) {
+
+	}
+	
+	protected void  setDefaultProperties() {
+		Optional<DefaultProperties> defaultPropertiesOpt = AopUtils.getAnnotation(proxyClass, DefaultProperties.class);
+		builder.defaultGroupKey(proxyClass.getSimpleName());
+		if (defaultPropertiesOpt.isPresent()) {
+			DefaultProperties defaultProperties = defaultPropertiesOpt.get();
+			builder.defaultProperties(defaultProperties);
+			if (StringUtils.isNotBlank(defaultProperties.groupKey())) {
+				builder.defaultGroupKey(defaultProperties.groupKey());
+			}
+			if (StringUtils.isNotBlank(defaultProperties.threadPoolKey())) {
+				builder.defaultThreadPoolKey(defaultProperties.threadPoolKey());
+			}
+		}
+		
+	}
+
+	protected  void setFallbackMethod() {
+		FallbackMethod fallbackMethod = MethodProvider.getInstance().getFallbackMethod(proxyClass, method);
+		if (fallbackMethod.isPresent()) {
+			fallbackMethod.validateReturnType(method);
+			builder.fallbackMethod(fallbackMethod.getMethod()).fallbackExecutionType(ExecutionType.getExecutionType(fallbackMethod.getMethod().getReturnType()));
+		}
+		
+	}
+	protected Method getBatchCommandMethod(HystrixCollapser collapser){
+		return getDeclaredMethod(proxyClass, collapser.batchMethod(), List.class);
+	}
+	protected Class<?> getFirstGenericParameter(Type type) {
+		return getFirstGenericParameter(type, 1);
+	}
+
+
+	protected Class<?> getFirstGenericParameter(final Type type, final int nestedDepth) {
```

Comment:
> Also: why do we need this method at all since it's used only for nestedDepth=1, can we simplify this ?

This method is called from two places one with depth=1 and the other one with depth=0. 

Regarding Accessors, I will make it static or final based on what ever is appropriate.